dcos/http.py

```diff
@@ -166,12 +172,24 @@ def request(method,
     if is_success(response.status_code):
         return response
     elif response.status_code == 401:
-        if auth_token is not None:
-            msg = ("Your core.dcos_acs_token is invalid. "
-                   "Please run: `dcos auth login`")
-            raise DCOSAuthenticationException(msg)
+        if prompt_login:
+            header_challenge_auth(dcos_url.geturl())
+            auth_token = config.get_config_val("core.dcos_acs_token", toml_config)
+            if auth_token and scheme_eq and netloc_eq:
+                auth = DCOSAcsAuth(auth_token)
+            else:
+                auth = None
+            response = _request(method, url, is_success, timeout,
```

Comment:
Ah sorry I misunderstood your comment before. You said `request` and I didn't notice the function was called `request` and thought you were referring to `_request`.

f673b3e